src/main/java/org/opensearch/neuralsearch/search/query/HybridQueryPhaseSearcher.java

```diff
@@ -44,6 +45,7 @@ public boolean searchWith(
         final boolean hasFilterCollector,
         final boolean hasTimeout
     ) throws IOException {
+        addTwoPhaseNeuralSparseQuery(query, searchContext);
```

Comment:
What we are doing in this function is not clear from the function name. Are we updating the query clause or we are updating the rescore context? 

Earlier also we discussed that hybrid queryphase searcher is not related to NeuralSparseQuery. Better way to do this exposing an interface from Opensearch core to override the search context.

Reply:
Now change the name to addRescoreContextFromNeuralSparseQuery.

As earlier discuss, we decided to remove this from hybridQueryPhaseSearcher asap, RFC draft in opensearch core  are on the way.